### ISSUE TITLE
Cannot start service with empty arguments

### ISSUE THREAD

**Issue by retep998** [mullvad/windows-service-rs#51]
When passing an empty array to `Service::start` Rust needs to be told the type as it cannot infer the type of an empty array. Before I could do `service.start::<OsString>(&[])?;`. In 0.3.0 `Service::start` was changed to use `impl AsRef<OsStr>` instead of an explicit type parameter which means I cannot tell Rust what the type is. Instead I have to do this:
```Rust
let arguments: &[OsString] = &[];
service.start(arguments)?;
```
Please reconsider.

**Comment by pronebird** [comment]
I guess ergonomics comes at a price. All in all I don't think it's worth a hassle.

There are alternatives that you may find more favourable:

1. The good old `as` cast

   ```
    service.start(&[] as &[&OsStr])?
    ```

1. Using `Default` trait implemented for empty arrays

    ```
    service.start(&<[&OsStr; 0]>::default())?
    ```

1. Perhaps not the most performant alternative but then maybe Rust optimises that out internally. I don't know, but here is another way via `Vec`:

     ````
     service.start(&Vec::<&OsStr>::new())?
    ````

1. Last but not least, you can define a static variable somewhere around and use it:

    ```
    const EMPTY_SERVICE_ARGUMENTS: &[&OsStr] = &[];
    
    service.start(EMPTY_SERVICE_ARGUMENTS)?
    ```

    We could perhaps provide a constant for that, i.e `Service::EMPTY_ARGUMENTS` or a method like `Service::start_without_arguments() but this seems like such an overkill at this point and then we would have to do that for ever occurrence where we use `impl T`.

**Comment by ErichDonGubler** [comment]
> Perhaps not the most performant alternative but then maybe Rust optimises that out internally. I don't know, but here is another way via `Vec`:
> 
> ```rust
> service.start(&Vec::<&OsStr\>::new())?
> ```

Empty `Vec`s instantiated with `Vec::new()` are [guaranteed not to allocate](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.new), so this is not really inefficient. 😃

**Comment by faern** [comment]
I'll close this as there are OK alternatives. However, we are making a breaking release shortly. So if someone is having strong opinions on this, now is the time to express them, so we can change it back to a generic, if people deem that to be better.

**Comment by faern** [comment]
This was fixed (reverted) and is part of the new `0.5.0` release. Ping @retep998
